### PR TITLE
Views & style

### PR DESCRIPTION
**Pull request by HFG43**
Hi there! I'm sending this pull request for you to review my code. Thanks in advance for your feedback!

Milestones 🎯:
📌 Generate views following `index` & `show` actions
📌 Give style to each view following wireframes design

🎖️ General requirements:
Gitflow
No linter errors
Ensure the use of best practices and Ruby style

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1 +1,9 @@
-<p>Here you will soon see the list of all users</p>
\ No newline at end of file
+<% @users.each do |user| %>
```
Comment (topeogunleye) [inline app/views/users/index.html.erb:1]: - [OPTIONAL] 📘 **Suggestion**: If you haven’t considered it already, using partials like `app/views/users/_user.html.erb` to render users is a best practice, making the code more modular and manageable. Here's how:
```erb
<%= render @users %>

**Review by topeogunleye — CHANGES_REQUESTED** [review]
Hi Hernán, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

Try-Again

> ## (Highlights) Good Points: 💪 

#### 🌐 **Controllers and Views**
- Your implementation of the `UsersController#index` action correctly fetches all the users and renders them in the corresponding view. 🎉
- The `index.html.erb` is well-structured, efficiently iterating over the `@users` instance variable to display user details. 🏆

#### 💬 **Comments & Posts Display**
- The post show page effectively displays the post with all comments listed, aligning with the wireframe. 📄

#### 🌈 **Styling (Nice to Have)**
- Though the focus was on functionality, your attention to styling and detail is commendable and contributes to an improved user experience. 🎨

#### 🌟 **Conclusion**
In conclusion, while you’ve made substantial progress and shown an understanding of the project requirements, there are some pivotal alterations and enhancements that are needed to fully align the project with the given specifications and best practices. 🛠️

The changes requested are crucial to refining your implementation and achieving a more polished and optimized outcome. Please review the specific comments and suggestions provided and make the necessary adjustments. 🔄

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### There was no chance to put the below comments inline:

You can also consider:
- Consider populating your `db/seed.rb` file with data. Here's how:
```erb
first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
second_user = User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.')
first_post = Post.create(author: first_user, title: 'Hello', text: 'This is my first post')
Comment.create(post: first_post, author: second_user, text: 'Hi Tom!' )
```

- For future pull requests, please notice that your branch name `views` does not follow the convention of being named after a feature or action. For example, you could name your branch `added-views`. This makes it easier to track the changes and the purpose of each branch.

After addressing these suggested improvements and any other highlighted issues, kindly resubmit the project for another review. I believe with these modifications, your project will not only meet the expectations but excel in terms of quality and adherence to best practices! 🌟

Remember, continuous improvement is key in coding. Keep up the great work, and feel free to ask for clarifications or additional support if needed. Happy coding! 🚀
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by HFG43** [comment]
@topeogunleye! Thanks for all your feedback and for taking the time to make it so clear and descriptive!

**Comment by topeogunleye** [comment]
> @topeogunleye! Thanks for all your feedback and for taking the time to make it so clear and descriptive!

Hello @HFG43,

You're very welcome! 🌟 I'm glad you found the feedback helpful and clear. It's fantastic to see your commitment and effort in understanding and applying the concepts. Should you have any more questions or need further clarification on any points, please feel free to ask. Keep up the great work, and I'm excited to see your continuous growth and more of your amazing projects in the future! 🚀

Happy Coding! 🌈

Best Regards,  
Temitope

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1 +1,21 @@
...
+  <h3>Bio</h3>
+  <p><%= @user.bio%></p>
+</section>
+
+<% @user.posts.each do |post|%>
+  <div class="container">
+    <h3><%= "Post # #{post.id}"%></h3>
+    <p><%= post.text %></p>
+    <h4><%= "Comments #{post.comments_counter}, Likes #{post.likes_counter}"%></h4>
+  </div>
```
[the remark this answers is not in the document]
Comment (KDlamini) [reply app/views/users/show.html.erb]: Thank you for the fantastic review @topeogunleye 💣 

This feature is well resolved ✔️

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -1 +1,24 @@
...
+</div>
+
+<% @posts.each do |post|%>
+  <div class="container">
+    <h3><%= "Post # #{post.id}"%></h3>
+    <p><%=post.text%></p>
+    <h4><%="Comments #{post.comments_counter}, Likes #{post.likes_counter}"%></h4>
+  </div>
+  <div class="container">
+    <% post.comments.each do |comment|%>
```
[the remark this answers is not in the document]
Comment (KDlamini) [reply app/views/posts/index.html.erb]: Thank you for the well-detailed review @topeogunleye 💯 

The `recent-comments` methods is well implemented ✔️

File: app/views/users/index.html.erb [hunk 4]
```diff
@@ -1 +1,9 @@
...
+  <div class="user_index_container">
+    <div class="container author_photo">Author Pic</div>
+    <div class="container user_index_content">
+      <h2><%= user.name %></h2>
+      <p><%= "Number of posts: #{user.posts_counter}"%></p>
+    </div>
+  </div>
```
[the remark this answers is not in the document]
Comment (KDlamini) [reply app/views/users/index.html.erb]: Thank you for the informative suggestion and encouragement @topeogunleye 

@HFG43 you have done an incredible job implementing the required changes. This functionality is well resolved ✔️

File: app/views/users/show.html.erb [hunk 5]
```diff
@@ -1 +1,21 @@
...
+<%end %>
+
+<div class="button_cont">
+  <button type="button">See all posts</button>
```
[the remark this answers is not in the document]
Comment (KDlamini) [reply app/views/users/show.html.erb]: The `See all posts` navigates to the Posts index page. Well resolved ✔️

**Review by KDlamini — APPROVED** [review]
### Hi @HFG43 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes for the Blog app views. 🎊 🥳 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!

## Highlights 🌞

✔️  Good PR title and description
✔️  Users controller `#index` action renders all users to their corresponding Views index page
✔️  The `#recent_post` method renders the user's 3 most recent posts.
✔️  `See all Post` link opens the user's post index as displayed on the wireframe
✔️  Post show page displays post with all comments

# Status: Approved ✔️ 

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by HFG43** [comment]
@KDlamini Hi KDlamini!! Many thanks for your review and feedback!!

**Pull request merged**